src/components/MyAccount/NewSettings/EmailForm.tsx

```diff
@@ -0,0 +1,207 @@
+      }
+    } catch (error) {
+      console.error("Error submitting emails:", error)
+    } finally {
+      setIsLoading(false)
+    }
+  }
+
+  return (
+    <Flex flexDir="row" alignItems="flex-start" width="100%">
+      <Flex gap="xs" alignItems={"center"} sx={{ paddingTop: "xs" }}>
+        <Icon name="communicationEmail" size="medium" />
+        <Text
+          size="body1"
+          sx={{
+            fontWeight: "500",
+            width: "256px",
+            marginBottom: 0,
+          }}
+        >
+          Email
+        </Text>
+      </Flex>
+      {isEditing ? (
+        <Flex flexDir={"column"}>
+          {tempEmails.map((email, index) => (
+            <Flex key={index} mb="s">
+              <TextInput
+                name={`email-${index}`}
+                value={email}
+                id={`email-text-input-${index}`}
+                labelText="Update email"
+                showLabel={false}
+                isInvalid={error && !validateEmail(email)}
+                invalidText="Please enter a valid email address."
+                onChange={(e) => handleInputChange(e, index)}
+                isRequired
+                isClearable
+                isClearableCallback={() => handleClearableCallback(index)}
+                sx={{ width: "300px" }}
+              />
+              {index !== 0 && (
+                <Button
+                  aria-label="Remove email"
+                  buttonType="text"
+                  id="remove-email-btn"
+                  onClick={() => handleRemoveEmail(index)}
+                >
+                  {" "}
+                  <Icon name="actionDelete" size="large" />
+                </Button>
+              )}
+            </Flex>
+          ))}
+          <Button
+            id="add-button"
+            buttonType="text"
+            onClick={handleAddEmail}
+            size="large"
+            sx={{
+              justifyContent: "flex-start",
+              width: "300px",
+              padding: "xxs",
+            }}
+          >
+            + Add an email address
+          </Button>
+        </Flex>
+      ) : (
+        <Flex flexDir="row" alignItems="flex-start">
+          <Flex flexDir="column" alignItems="flex-start">
+            {emails.map((email, index) => (
+              <Text key={index} sx={{ paddingTop: "xs", marginBottom: "xs" }}>
+                {email}{" "}
+                {index === 0 && <span style={{ color: "gray" }}>(P)</span>}
+              </Text>
+            ))}
+          </Flex>
+          <Button
+            id="edit-email-button"
+            buttonType="text"
+            onClick={() => setIsEditing(true)}
+            sx={{ paddingLeft: "xs", paddingRight: "xs", marginLeft: "xxl" }}
+          >
+            <Icon name="editorMode" align="left" size="medium" />
+            Edit
+          </Button>
+        </Flex>
+      )}
+      {isEditing && (
+        <Flex justifySelf="flex-end" sx={{ marginLeft: "auto" }}>
```

Comment:
Can you try using the `ButtonGroup` component? If you already did, did it not work? It has a default gap between the buttons so you shouldn't need the margins in the first button.